src/components/sendV2/form/form.js

```diff
@@ -0,0 +1,320 @@
+          showSuggestions: false,
+        },
+        amount: {
+          error: false,
+          value: '',
+          feedback: '',
+        },
+        reference: {
+          error: false,
+          value: '',
+          feedback: '',
+        },
+      },
+    };
+
+    this.loaderTimeout = null;
+
+    this.validateAmountAndReference = this.validateAmountAndReference.bind(this);
+    this.onAmountOrReferenceChange = this.onAmountOrReferenceChange.bind(this);
+    this.onSelectedAccount = this.onSelectedAccount.bind(this);
+    this.onInputChange = this.onInputChange.bind(this);
+    this.validateBookmark = this.validateBookmark.bind(this);
+    this.onGoNext = this.onGoNext.bind(this);
+  }
+
+  onInputChange({ target }) {
+    this.setState({
+      fields: {
+        ...this.state.fields,
+        [target.name]: {
+          ...this.state.fields[target.name],
+          value: target.value,
+        },
+      },
+    });
+  }
+
+  // eslint-disable-next-line max-statements
+  validateBookmark() {
+    const { followedAccounts } = this.props;
+    let recipient = this.state.fields.recipient;
+    let isAccountValid = '';
+    let isAddressValid = '';
+
+    if (followedAccounts.length && recipient.value !== '') {
+      isAccountValid = followedAccounts
+        .find(account => account.title.toLowerCase() === recipient.value.toLowerCase()) || false;
+      isAddressValid = regex.address.test(recipient.value);
+    } else {
+      isAddressValid = recipient.value.match(regex.address);
+    }
+
+    // istanbul ignore else
+    if (isAddressValid) {
+      recipient = {
+        ...this.state.recipient,
+        address: recipient.value,
+        selected: false,
+        error: false,
+        feedback: '',
+        showSuggestions: false,
+      };
+    }
+
+    // istanbul ignore else
+    if (isAccountValid) {
+      recipient = {
+        ...this.state.recipient,
+        address: isAccountValid.address,
+        title: isAccountValid.title,
+        balance: isAccountValid.balance,
+        selected: true,
+        error: false,
+        feedback: '',
+        showSuggestions: false,
+      };
+    }
+
+    // istanbul ignore else
+    if (!isAccountValid && !isAddressValid && recipient.value) {
+      recipient = {
+        ...this.state.recipient,
+        address: '',
+        balance: '',
+        error: true,
+        feedback: 'Provide a correct wallet address or a name of a followed account',
+        selected: false,
+        title: '',
+        showSuggestions: true,
+      };
+    }
+
+    // istanbul ignore else
+    if (recipient.value === '') {
+      recipient = {
+        ...this.state.recipient,
+        address: '',
+        balance: '',
+        error: false,
+        feedback: '',
+        selected: false,
+        title: '',
+        showSuggestions: false,
+      };
+    }
+
+    this.setState({
+      fields: {
+        ...this.state.fields,
+        recipient: {
+          ...this.state.fields.recipient,
```

Comment:
Since it's being destructured in each condition above, shouldn't need this one here.